File: examples/scala/polymorphic-app/src/main/scala/logging/ConsoleAppender.scala
package logging

import com.typesafe.config.ConfigTypeName

import scala.beans.BeanProperty

@ConfigTypeName("console")
class ConsoleAppender extends Appender {

    @BeanProperty var target: String = _

    override def log(event: String) {
        printf("Logging to '%s' using console logger: %s\n", target, event)
    }
}

File: examples/java/polymorphic-app/src/main/java/logging/ConsoleAppender.java
package logging;

public class ConsoleAppender implements Appender {

    private String target;

    public String getTarget() {
        return target;
    }

    public void setTarget(String target) {
        this.target = target;
    }

    @Override
    public void log(String event) {
        System.out.printf("Logging to '%s' using console logger: %s\n",
            target, event);
    }
}
